client/src/modules/detailedView/DetailedViewHeader/DetailedViewHeader.tsx

```diff
@@ -27,7 +37,16 @@ export default class DetailedViewHeader extends React.Component<Props> {
           <span className={styles.badge}>
             <img className={styles.icon} src={iconSrc} alt="Icon" />
           </span>
-          <h1 className={styles.name}>{model.name}</h1>
+          <h1 onClick={this.toggleModelNamePopover} className={styles.name} id={`${this.props.modelType}-header-name`}>{model.name}</h1>
+          <EditLeadFieldPopover
```

Comment:
I think we should rename this component now, it is now used for updating names of different models